packages/spatial/src/xr/XRSessionFunctions.test.ts

```diff
@@ -325,16 +333,155 @@ describe('onSessionEnd', () => {
     expect(result).toBe(Expected)
   })
 
+  it('should call `dispatchAction` with XRAction.sessionChanged{active:false}', () => {
+    const Expected = false
+    // Run and Check the result
+    onSessionEnd()
+    const result = getIncomingAction(XRAction.sessionChanged.type)
+    assert(result)
+    expect(typeof result).toBe('object')
+    // expect(result.active).toBe(Expected)
+  })
+}) //:: onSessionEnd
+
+describe('getReferenceSpaces', () => {
+  beforeEach(async () => {
+    createEngine()
+    await mockEmulatedXREngine()
+  })
+
+  afterEach(() => {
+    destroyEmulatedXREngine()
+    destroyEngine()
+  })
+
+  it("should call `@param xrSession`.requestReferenceSpace with a value of `'local-floor'`", () => {
+    const Expected = 'local-floor'
+    // Setup the data as expected
+    const xrSession = getMutableState(XRState).session.value as XRSession
+    const requestReferenceSpaceSpy = vi.spyOn(xrSession, 'requestReferenceSpace')
+    // Run and Check the result
+    getReferenceSpaces(xrSession)
+    expect(requestReferenceSpaceSpy).toHaveBeenCalled()
+    expect(requestReferenceSpaceSpy).toHaveBeenCalledWith(Expected)
+  })
+
+  it("should call `@param xrSession`.requestReferenceSpace with a value of `'viewer'`", () => {
+    const Expected = 'viewer'
+    // Setup the data as expected
+    const xrSession = getMutableState(XRState).session.value as XRSession
+    const requestReferenceSpaceSpy = vi.spyOn(xrSession, 'requestReferenceSpace')
+    // Run and Check the result
+    getReferenceSpaces(xrSession)
+    expect(requestReferenceSpaceSpy).toHaveBeenCalled()
+    expect(requestReferenceSpaceSpy).toHaveBeenCalledWith(Expected)
+  })
+}) //:: getReferenceSpaces
+
+describe.only('setupXRSession', () => {
+  beforeEach(async () => {
+    createEngine()
+    mockSpatialEngine()
+    WebXREventDispatcher.instance.dispatchEvent({
+      type: POLYFILL_ACTIONS.DEVICE_INIT,
+      detail: { stereoEffect: false, device: DeviceDefinitions.Default }
```

Comment:
`device` should be `deviceDefinition`

Reply:
Good catch, thanks!
Fixed with 81bce7e0aa87a77edee4ea2238475a826b52661e